src/java.compiler/share/classes/javax/lang/model/AnnotatedConstruct.java

```diff
@@ -45,14 +45,14 @@
  * <em>type annotation</em>.
  *
  * The terms <em>directly present</em>, <em>present</em>,
- * <em>indirectly present</em>, and <em>associated </em> are used
+ * <em>indirectly present</em>, and <em>associated</em> are used
  * throughout this interface to describe precisely which annotations,
  * either declaration annotations or type annotations, are returned by
  * the methods in this interface.
  *
  * <p>In the definitions below, an annotation <i>A</i> has an
  * annotation interface <i>AI</i>. If <i>AI</i> is a repeatable annotation
- * interface, the type of the containing annotation is <i>AIC</i>.
+ * interface, the type of the container annotation is <i>AIC</i>.
```

Comment:
"the annotation interface of the container annotation is *AIC*"
or
"the containing annotation interface is *AIC*"

To align with the JLS.

Reply:
"Type" is appropriate here and in some other locations in the `AnnotatedConstruct` javadoc. I paid attention to the difference between "contain-ing annotation interface" and "contain-er annotation".

@dansmithcode, please correct me if I'm wrong.

Reply:
For convenience, here's the rendered documentation in question for JDK 23: https://download.java.net/java/early_access/jdk23/docs/api/java.compiler/javax/lang/model/AnnotatedConstruct.html#:~:text=containing

Reply:
The "type" terminology is outdated and is already migrated to "class or interface" or the applicable subtype elsewhere: https://bugs.openjdk.org/browse/JDK-8257636 and commit 952dc704

Reply:
> The "type" terminology is outdated and is already migrated to "class or interface" or the applicable subtype elsewhere: https://bugs.openjdk.org/browse/JDK-8257636 and commit [952dc70](https://github.com/openjdk/jdk/commit/952dc704024d44c8f37449382fe769320f9dad1c)

@liach, I respect you as a fellow JLS reader. Not only am I aware of that JBS issue, but I also remember the respective PR being reviewed.

Unsurprisingly, I had a similar knee-jerk reaction to the word "type". So, before publishing this PR, I asked a JLS expert on its use in this context. He replied that although in general "type" should not to be conflated with "interface"/"class", in this context it is reasonable to use "type".

However, if we do decide to abandon "type", we need to inspect every use of it in this class very carefully. The closest terminology that I could find is JLS 9.7.1

> The `TypeName` specifies the annotation interface corresponding to the annotation. The annotation is said to be "of" that interface.

and JLS 9.6.3

> An annotation interface `A` is repeatable if its declaration is (meta-)annotated with an `@Repeatable` annotation ([§9.6.4.8](https://docs.oracle.com/javase/specs/jls/se22/html/jls-9.html#jls-9.6.4.8)) whose value element indicates a containing annotation interface of `A`.

Note how slightly dizzying the terminology is.

Reply:
Separately, maybe JLS hasn't done its homework in full? See 9.6.4.1. `@Target`:
> It is a compile-time error if the same enum constant appears more than once in the value element of an annotation of type `java.lang.annotation.Target`.

Reply:
Yes, that use of "type" in 9.6.4.1. feels like an overlook.

Reply:
Chen, I suggest we integrate this PR as is but ask JLS experts/authors on the issue. In fact, I've already asked. I suspect that JLS 9.6.4.1 is germane to this discussion. If JLS changes "type" to "interface" in JLS 9.6.4.1, we could change these occurrences of "type" to "interface" in `AnnotatedConstruct` too.

Reply:
There's really nothing wrong with talking about the type of an annotation, just like you can talk about the type of a variable. If it's the most intuitive way to express what you're trying to say, go for it!